gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingExtractor.java

```diff
@@ -0,0 +1,209 @@
+	    }
+
+      @Override
+	    public ComparableWatermark getWatermark() {
+	      return _lwm;
+	    }
+
+      @Override
+      public short calculatePercentCompletion(Watermark lowWatermark, Watermark highWatermark) {
+        return 0;
+      }
+
+      @Override
+      public JsonElement toJson() {
+        return null;
+      }
+
+      @Override
+      public int compareTo(CheckpointableWatermark o) {
+        Preconditions.checkArgument(o instanceof KafkaWatermark);
+        KafkaWatermark ko = (KafkaWatermark)o;
+        Preconditions.checkArgument(this._topicPartition.equals(ko._topicPartition));
+        return this._lwm.compareTo(ko._lwm);
+      }
+
+      @Override
+      public boolean equals(Object obj) {
+        if (obj == null)
+          return false;
+        Preconditions.checkArgument(obj instanceof KafkaWatermark);
+        return this.compareTo((CheckpointableWatermark)obj) == 0;
+      }
+
+      @Override
+      public int hashCode() {
+        final int prime = 31;
+        return _topicPartition.hashCode() * prime + _lwm.hashCode();
+      }
+
+      public TopicPartition getTopicPartition() {return _topicPartition;};
+
+      public LongWatermark getLwm() {return  _lwm;};
+    }
+
+    private final KafkaSchemaRegistry<String, String> _kafkaSchemaRegistry;
+    private Consumer<String, byte[]> _consumer;
+    private TopicPartition _partition;
+    private Iterator<ConsumerRecord<String, byte[]>> _records;
+    long _rowCount = 0;
+
+    public KafkaSimpleStreamingExtractor(WorkUnitState state) {
+      super(state);
+      this._kafkaSchemaRegistry = new SimpleKafkaSchemaRegistry(state.getProperties());
+      this._consumer = KafkaSimpleStreamingSource.getKafkaConsumer(state);
+      this.closer.register(_consumer);
+      this._partition = new TopicPartition(state.getProp(KafkaSimpleStreamingSource.TOPIC_NAME),
+                                           state.getPropAsInt(KafkaSimpleStreamingSource.PARTITION_ID));
+      _consumer.assign(Collections.singletonList(_partition));
+      OffsetAndMetadata offset = _consumer.committed(_partition);
+      if (offset == null)
+        _consumer.seekToBeginning(_partition);
+      else
+        _consumer.seek(_partition, offset.offset());
+    }
+
+    /**
+    * Get the schema (metadata) of the extracted data records.
+    *
+    * @return the Kafka topic being extracted
+    * @throws IOException if there is problem getting the schema
+    */
+    @Override
+    public String getSchema() throws IOException {
+      try {
+        return this._kafkaSchemaRegistry.getLatestSchemaByTopic(this._partition.topic());
+      } catch (SchemaRegistryException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @Override
+    public List<Tag<?>> generateTags(State state) {
+      List<Tag<?>> tags = super.generateTags(state);
+      tags.add(new Tag<>("kafkaTopic", state.getProp(KafkaSimpleStreamingSource.TOPIC_WHITELIST)));
```

Comment:
right now, the assumption is its a single topic. I'll try to rationalize with the SimpleKafkaSource but that depends on KafkaWrapper and the new api which hasnt been implemented yet. Eventually this will move to using KafkaWrapper.